test/cpp_api_parity/sample_module.py

```diff
@@ -1,124 +1,106 @@
             self.param.fill_(1)
-            self.buffer.fill_(1)
-            self.attr = 10
-            if not self.has_parity:
-                self.param.add_(10)
-                self.buffer.add_(10)
-                self.attr += 90
 
     def forward(self, x):
         submodule_forward_result = self.submodule(x) if hasattr(self, 'submodule') else 0
-        if not self.has_parity:
-            return x + self.param * 4 + submodule_forward_result + 3
-        else:
+        if self.has_parity:
             return x + self.param * 2 + submodule_forward_result
+        else:
+            return x + self.param * 4 + submodule_forward_result + 3
+
+torch.nn.SampleModule = SampleModule
 
 SAMPLE_MODULE_CPP_SOURCE = """\n
 namespace torch {
-namespace nn{
+namespace nn {
 struct C10_EXPORT SampleModuleOptions {
-  SampleModuleOptions(bool has_submodule) : has_submodule_(has_submodule) {}
+  SampleModuleOptions(bool has_parity, bool has_submodule) : has_parity_(has_parity), has_submodule_(has_submodule) {}
+
+  TORCH_ARG(bool, has_parity);
   TORCH_ARG(bool, has_submodule);
-  TORCH_ARG(int64_t, int_option) = 0;
-  TORCH_ARG(double, double_option) = 0.1;
-  TORCH_ARG(bool, bool_option) = false;
-  TORCH_ARG(std::string, string_option) = "0";
-  TORCH_ARG(torch::Tensor, tensor_option) = torch::zeros({1});
-  TORCH_ARG(ExpandingArray<2>, int_or_tuple_option) = 0;
 };
 
 struct C10_EXPORT SampleModuleImpl : public torch::nn::Cloneable<SampleModuleImpl> {
-  SampleModuleImpl(bool has_submodule) : SampleModuleImpl(SampleModuleOptions(has_submodule)) {}
   explicit SampleModuleImpl(SampleModuleOptions options) : options(std::move(options)) {
     if (options.has_submodule()) {
-      submodule = register_module("submodule", std::make_shared<SampleModuleImpl>(false));
+      submodule = register_module(
+        "submodule",
+        std::make_shared<SampleModuleImpl>(SampleModuleOptions(options.has_parity(), false)));
     }
     reset();
   }
   void reset() {
-    attr = 10;
     param = register_parameter("param", torch::ones({3, 4}));
-    buffer = register_buffer("buffer", torch::ones({4, 5}));
   }
   torch::Tensor forward(torch::Tensor x) {
     return x + param * 2 + (submodule ? submodule->forward(x) : torch::zeros_like(x));
   }
   SampleModuleOptions options;
   torch::Tensor param;
-  torch::Tensor buffer;
-  int attr;
   std::shared_ptr<SampleModuleImpl> submodule{nullptr};
 };
 
 TORCH_MODULE(SampleModule);
-}
-}
+} // namespace nn
+} // namespace torch
 """
 
 module_tests = [
     dict(
         module_name='SampleModule',
         desc='has_parity',
         constructor_args=(True, True),
-        cpp_constructor_args='(true)',
+        cpp_constructor_args='torch::nn::SampleModuleOptions(true, true)',
         input_size=(3, 4),
+        cpp_input_args=['torch::randn({3, 4})'],
         has_parity=True,
     ),
     dict(
         fullname='SampleModule_no_parity',
         constructor=lambda: SampleModule(False, True),
```

Comment:
Nit: this is easier to read if we pass False, True as keyword args: `lambda: SampleModule(has_parity=False, has_submodule=True)`. Otherwise the reader has to scroll up. Same for the next dict in this file